lib/DeveloperStatsCommander.js

```diff
@@ -0,0 +1,234 @@
+
+                    var statsTimestamp = weeklyStats.weeks[w].w;
+                    var thisWeekTimestamp = moment().startOf('week').add(1, 'hour').format('X');
+
+                    if (statsTimestamp != thisWeekTimestamp) {
+                        continue;
+                    }
+
+                    if (typeof digestedContributorStats [weeklyStats.author.login] == 'undefined') {
+                        digestedContributorStats [weeklyStats.author.login] = {};
+                    }
+
+                    if (typeof digestedRepoStats [repo] == 'undefined') {
+                        digestedRepoStats [repo] = {};
+                    }
+
+                    var entry = {
+                        commits: weeklyStats.weeks[w].c,
+                        added : weeklyStats.weeks[w].a,
+                        deleted : weeklyStats.weeks[w].d,
+                        changed : weeklyStats.weeks[w].a + weeklyStats.weeks[w].d
+                    };
+
+                    if (entry.commits == 0) {
+                        continue;
+                    }
+
+                    digestedContributorStats [weeklyStats.author.login][repo] = entry;
+                    digestedRepoStats [repo][weeklyStats.author.login] = entry;
+                }
+            }
+        }
+
+        var repoStats = this.snitch (digestedRepoStats);
+        var contributorStats = this.snitch (digestedContributorStats);
+
+        console.log (repoStats);
+        console.log (contributorStats);
+
+        if (process.env.SLACK_ENABLE == 'true') {
+            this.slack (repoStats);
+            this.slack (contributorStats);
+        }
+
+        return;
+    }
+
+    DeveloperStatsCommander.prototype.slack = function (message) {
+
+        var sh = 'curl -X POST --data-urlencode \'payload={"username": "' + process.env.SLACK_USERNAME + '", "text": "'+ message + '", "icon_emoji": "' + process.env.SLACK_EMOJI + '"}\' ' + process.env.SLACK_URL;
+
+        exec(sh, function(error, stdout, stderr) {
+            if (error) {
+                throw error;
+            }
+        });
+    }
+
+    DeveloperStatsCommander.prototype.snitch = function (digested) {
+
+        var result = '';
+        var zeroes = [];
+
+        for (var p in digested) {
+
+            var totalCommits = 0;
+            var totalChanges = 0;
+            var line = '';
+            var notes = '';
+
+
+            for (var name in digested [p]) {
+
+                totalCommits += digested[p][name].commits;
+                totalChanges += digested[p][name].changed;
+
+                if (notes != '') {
+                    notes += ', ';
+                }
+
+                notes += name + ': ' + digested[p][name].commits;
+
+            }
+
+            if (totalCommits == 0) {
+                zeroes.push (p);
+                continue;
+            }
+
+            notes = '(' + notes + ')';
+
+            line += '`' + ('   ' + totalCommits).substr(-3) + '` ';
```

Comment:
Maybe we should use some kind of templating engine for this like `mustache`. Inline template is bad, even in JavaScript :wink: